addons/im_livechat/static/src/legacy/widgets/public_livechat_window/public_livechat_window.js

```diff
@@ -179,8 +179,13 @@ const PublicLivechatWindow = Widget.extend({
      * @private
      * @param {Object} messageData
      */
-    _postMessage(messageData) {
-        this.trigger_up('post_message_chat_window', { messageData });
+    async _postMessage(messageData) {
+        try {
+            await this.messaging.publicLivechatGlobal.livechatButtonView.sendMessage(messageData);
+        } catch (reason) {
+            reason.event.preventDefault();
+            return this.messaging.publicLivechatGlobal.livechatButtonView.sendMessage(messageData); // try again just in case
```

Comment:
What if it succeeds the second time? You return whereas you don't do it when it works the first time. This asymmetry of behaviour seems a bit odd to me, I feel like there is something wrong.

Reply:
should have been `await` too, change done